File: 2021/day1/day1.py
import os

filename = "/home/looten/workspace/advent_of_code/2021/day1_input.txt"

def readfile():
    prev_val = -1
    cnt = 0

    with open(filename) as f:
        lines = f.readlines()
        array = []

        for i in lines:
            if type(i):
                array.append(i)

        n = len(array)
        k = 3

        for i in range(n-k+1):
            current_sum = 0

            for j in range(k):
                current_sum = current_sum + int(array[i + j])

            if prev_val == -1:
                prev_val = current_sum
                continue

            if current_sum > prev_val:
                cnt = cnt + 1

            prev_val = current_sum

    print(cnt)

if __name__ == "__main__":
    readfile()

File: 2022/day1.go
package main

import (
	"fmt"
	"os"
	"bufio"
	"strconv"
)

func check(err error) {
	if err != nil {
		panic(err)
	}
}

func sum(data []int) int {
	sum := 0
	for _, a := range data {
		sum += a
	}
	return sum;
}

func find_max(values []int) int {
	max := values[0]
	for _, v := range values {
		if (v > max) {
			max = v
		}
	}
	return max
}

func read_file(){
	path := "/home/looten/workspace/advent_of_code/2022/input_data.txt"
	// path := "/home/looten/workspace/advent_of_code/2022/test_data.txt"
	file, err := os.Open(path)
	check(err)

	defer file.Close()
	scanner := bufio.NewScanner(file)
    
	var values []int
	var tmp []int
	//idx := 0
    for scanner.Scan() {
		str := scanner.Text()
		if len(str) > 0 {
			// found value on row
			value, err := strconv.Atoi(str)
			check(err)
			tmp = append(tmp, value)
		} else {
			// empty row
			tot_sum := 0
			tot_sum = sum(tmp)
			values = append(values, tot_sum)
			values = append(values, -1)
			tmp = nil
		}
    }
	fmt.Println(find_max(values))
}

func main(){
	read_file()
}